drivers/SmartThings/zigbee-button/src/zigbee-multi-button/robb/init.lua

```diff
@@ -0,0 +1,226 @@
+  device:send(PowerConfiguration.attributes.BatteryPercentageRemaining:configure_reporting(device, 30, 21600, 1))
+
+  for endpoint = 1, endpoints do
+    device:send(device_management.build_bind_request(device, Level.ID, driver.environment_info.hub_zigbee_eui, endpoint))
+    device:send(device_management.build_bind_request(device, OnOff.ID, driver.environment_info.hub_zigbee_eui, endpoint))
+  end
+
+  device:send(OnOff.attributes.OnOff:configure_reporting(device, 0, 600, 1))
+
+  device:send(Basic.attributes.DeviceEnabled:write(device, true))
+  if not device:get_field('is_group_configured') then
+    -- Configure adding hub to group once
+    driver:add_hub_to_zigbee_group(0xE902)
+    -- Add two more groups if it is the 8x wall switch
+    if endpoints == 4 then
+      driver:add_hub_to_zigbee_group(0xE903)
+      driver:add_hub_to_zigbee_group(0xE904)
+    end
+
+    device:set_field('is_group_configured', true, { persist = true })
+  end
+end
+
+local function added_handler(self, device)
+  local supported_button_values
+  local number_of_buttons
+  local comp_id
+
+  for _, comp in pairs(device.profile.components) do
+    comp_id = comp.id
+
+    -- Special handeling of ON buttons (left column)
+    if comp_id == "button1" or comp_id == "button3" or comp_id == "button5" or comp_id == "button7" then
+      supported_button_values = { "pushed", "up_hold" }
+      number_of_buttons = 1
+    -- Special handling of OFF buttons (right column)
+    elseif comp_id == "button2" or comp_id == "button4" or comp_id == "button6" or comp_id == "button8" then
+      supported_button_values = { "pushed", "down_hold" }
+      number_of_buttons = 1
+    -- Handling of main button capability
+    elseif comp_id == "main" then
+      supported_button_values = { "pushed", "up_hold", "down_hold" }
+      number_of_buttons = WIRELESS_REMOTE_FINGERPRINTS[device:get_model()].buttons
+    end
+
+    device:emit_component_event(comp,
+      capabilities.button.supportedButtonValues(supported_button_values, { visibility = { displayed = false } }))
+    device:emit_component_event(comp,
+      capabilities.button.numberOfButtons({ value = number_of_buttons }, { visibility = { displayed = false } }))
+  end
+  device:emit_event(capabilities.button.button.pushed({ state_change = false }))
+  device:send(PowerConfiguration.attributes.BatteryPercentageRemaining:read(device))
+end
+
+local battery_perc_attr_handler = function(driver, device, value, zb_rx)
+  device:emit_event(capabilities.battery.battery(utils.clamp_value(value.value, 0, 100)))
+end
+
+-- Map components to endpoints
+local function component_to_endpoint(device, component_id)
+  local COMP_MAP = {
+    ["button1"] = 1,
+    ["button3"] = 2,
+    ["button5"] = 3,
+    ["button7"] = 4,
+    ["button2"] = 1,
+    ["button4"] = 2,
+    ["button6"] = 3,
+    ["button8"] = 4
+  }
+
+  if COMP_MAP[component_id] ~= nil then
+    return COMP_MAP[component_id]
+  else
+    return 1
+  end
+end
```

Comment:
I think this is unused? Since you're not receiving any commands that have to be sent to a specific component, you don't need this mapping. This is for things like switches where you need to be able to translate "turn switch1 on" to "send on command to endpoint 2" and such

Reply:
You're right it's not necessary. It was added to face some upcoming battery trouble after adding endpoint_to_component() but now it is not used anymore. Removed with [8fb92ba](https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/pull/604/commits/8fb92ba337ac7bcc1e11c99e0dd85f5e5295cac0)